From 565f1cfcf0eda16fbe97d494eb8ea2b1fd7d8072 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 31 Mar 2015 12:50:21 +0200
Subject: [PATCH 4/4] udevng: retrieve network interface for MBM from env when
 not found

In some scenarios sysfs doesn't report us a network interface to use for the MBM modem.
For such cases we allow the user to specify OFONO_MBM_NETWORK_INTERFACE=<interface name>
to workaround this (however they could still fix the kernel).
---
 ofono/plugins/udevng.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ofono/plugins/udevng.c b/ofono/plugins/udevng.c
index 09740ae..e3b7f9e 100644
--- a/ofono/plugins/udevng.c
+++ b/ofono/plugins/udevng.c
@@ -126,6 +126,10 @@ static gboolean setup_mbm(struct modem_info *modem)
 	if (mdm == NULL || app == NULL)
 		return FALSE;
 
+	if (network == NULL)
+		network = getenv("OFONO_MBM_NETWORK_INTERFACE");
+
+
 	DBG("modem=%s data=%s network=%s gps=%s", mdm, app, network, gps);
 
 	ofono_modem_set_string(modem->modem, "ModemDevice", mdm);
-- 
2.1.4

